src/solvers/refinement/string_refinement.cpp

```diff
@@ -1067,7 +1067,7 @@ static exprt substitute_array_access(
   const bool left_propagate)
 {
   return left_propagate ? interval_sparse_arrayt(expr).to_if_expression(index)
-                        : sparse_arrayt(expr).to_if_expression(index);
+                        : sparse_arrayt::to_if_expression(expr, index);
```

Comment:
@romainbrenguier This is the code that does explicitly distinguish `sparse_arrayt` vs `interval_sparse_arrayt`